graphql_compiler/compiler/emit_sql.py

```diff
@@ -239,19 +460,118 @@ def start_global_operations(self):
+            outer_vertex_alias: the sqlalchemy alias corresponding to the vertex outside the fold
+            join_descriptor: JoinDescriptor from the schema describing how to join
+            outer_vertex_alias and output_vertex_alias.
+        """
+        # basic info about the outer vertex and output vertex
+        output_vertex_alias = self._sql_schema_info.vertex_name_to_table[
+            self._ir.query_metadata_table.get_location_info(fold_scope_location).type.name
+        ].alias()
+
+        join_descriptor = self._get_fold_join_descriptor_from_current_classname(fold_scope_location)
+
+        return output_vertex_alias, join_descriptor
+
+    def fold(self, fold_scope_location):
+        """Begin execution of a Fold Block."""
+        # This method traverses the first vertex in the fold, and if applicable produces the
+        # output and group by clauses.
+        if self._in_fold:
+            raise AssertionError(u'Fold block {} entered while inside another '
+                                 u'fold block at current location {}.'
+                                 .format(fold_scope_location, self._current_location_info))
+        # begin the fold
+        self._in_fold = True
+
+        # 1. get fold metadata
+        # location of vertex that is folded on
+        self._fold_vertex_location = fold_scope_location
+        outer_alias = self._current_alias.alias()
+
+        # 2. get information on the folded vertex and its edge to the outer vertex
+        fold_vertex_alias, join_descriptor = self._get_fold_join_info(fold_scope_location)
+
+        # 3. initialize fold object
+        self._fold = SQLFoldObject(outer_alias, join_descriptor)
+
+        # 4. add join information for this traversal to the fold object
+        self._fold.visit_traversed_vertex(join_descriptor, outer_alias, fold_vertex_alias)
+
+        # 5. add output columns to fold object
+        self._fold.visit_output_vertex(fold_vertex_alias, fold_scope_location, join_descriptor,
+                                       self._all_folded_outputs)
+
+    def unfold(self):
+        """Complete the execution of a Fold Block."""
+        fold_subquery, from_clause, outer_vertex = self._fold.end_fold(self._alias_generator,
+                                                                       self._from_clause,
+                                                                       self._current_alias)
+        subquery_key = (self._fold_vertex_location.base_location.query_path,
+                        self._fold_vertex_location.fold_path)
+
+        # Replaces the table first placed in the dict during the MarkLocation
+        # following the open of the fold scope. It needs to be replaced because while
+        # inside the fold scope, locations referring to subquery_key access columns
+        # from the output vertex table, while after the Unfold, those locations refer to the
+        # subquery constructed for the fold.
```

Comment:
nit: `Replaces` -> `Replace`, `open` -> `opening` or `start`

I think the second sentence is a bit wordy/confusing here. It might be clearer if you break it up into more than 1 sentence.